Prevent winit from overriding LSUIElement in package manifests

purajit opened the pull request:
- [x] Tested on all platforms changed
- [x] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

madsmtm commented on src/platform/macos.rs:
```diff
@@ -257,11 +257,10 @@ impl WindowExtMacOS for dyn Window + '_ {
 /// Corresponds to `NSApplicationActivationPolicy`.
-#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum ActivationPolicy {
     /// Corresponds to `NSApplicationActivationPolicyRegular`.
-    #[default]
     Regular,
```
This is a breaking change, and will prevent us from releasing it in a minor version. Either don't do this, or make the bigger change of also changing `with_activation_policy` to take an option.

purajit replied:
Ah you got to this too fast - didn't think you'd see my draft PR. Yeah, this is a remnant while I was trying a couple things. Was just removing it, don't wish to actually do this.

madsmtm commented on src/platform/macos.rs:
```diff
@@ -438,7 +437,7 @@ pub trait EventLoopBuilderExtMacOS {
 impl EventLoopBuilderExtMacOS for EventLoopBuilder {
     #[inline]
     fn with_activation_policy(&mut self, activation_policy: ActivationPolicy) -> &mut Self {
```
Please document the new behaviour, so that it's clear that calling `with_activation_policy` will _overwrite_ the value.

madsmtm reviewed (changes requested):
Looks good with comments resolved, thanks!

purajit commented:
@madsmtm addressed your comments, and thanks for the quick review! I've tested with a branch 
off of v0.30.4 since v0.30.5 (clean) breaks the application I'm testing it through. Do you think
this is good enough? If not, I can try doing a test with the example tomorrow.

madsmtm reviewed (approved):
Looks good, thanks! And yeah, your testing is good enough.

purajit commented:
Awesome, thanks!
